custom_components/ha_washdata/history_import.py

```
@@ -0,0 +1,840 @@
+            skipped=idle + gated,
+            parse_report=parse_report,
+        )
+    except Exception as exc:  # pylint: disable=broad-exception-caught
+        _LOGGER.debug("History-import scan build failed: %s", exc)
+        return {"error": "scan_failed"}
+
+
+# ─── Persistence ──────────────────────────────────────────────────────────────
+
+
+def dedup_key(start_time: Any, duration: Any) -> tuple[int, int] | None:
```

Ruff (0.16.1)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `start_time`

(ANN401)